partner_communication_switzerland/models/res_partner_zoom_attendee.py

```diff
@@ -85,23 +92,33 @@ def notify_user(self):
                 note=self.optional_message,
                 user_id=user_id)
 
-    def send_confirmation(self):
-        confirmation_config = self.env.ref(
-            "partner_communication_switzerland"
-            ".config_onboarding_zoom_registration_confirmation"
-        )
-        short_notice_config = self.env.ref(
-            "partner_communication_switzerland.config_onboarding_zoom_link")
-        for attendee in self:
-            object_id = attendee.id
-            config = confirmation_config
-            if attendee.zoom_session_id.date_start <= datetime.now() + relativedelta(
-                    days=2):
-                config = short_notice_config
-                object_id = attendee.zoom_session_id.id
-            self.env["partner.communication.job"].create({
-                "partner_id": attendee.partner_id.id,
-                "config_id": config.id,
-                "object_ids": object_id
-            })
-        return True
+    def send_communication(self, config_name):
+        config_id = self.env.ref(config_name.value).id
+        partner_id = self.partner_id.id
+
+        # avoid sending this twice communication
+        if config_name == ZoomCommunication.LINK:
+            if self.link_received:
+                return
+            else:
+                self.link_received = True
+
+        if config_name in [ZoomCommunication.REMINDER, ZoomCommunication.LINK]:
+            object_id = self.zoom_session_id.id
+        elif config_name in [ZoomCommunication.REGISTRATION]:
+            object_id = self.id
+        else:
+            object_id = None
+
+        return self.env["partner.communication.job"].create({
+            "config_id": config_id,
+            "partner_id": partner_id,
+            "object_ids": object_id,
+        })
+
+    @api.one
```

Comment:
api.one is deprectated and should not be used. but if you are sure you always call this method with a single record you can start the method with `self.ensure_one`